llvm/lib/Transforms/InstCombine/InstCombineNegator.cpp

```diff
@@ -309,6 +310,9 @@ std::array<Value *, 2> Negator::getSortedOperandsOfBinOp(Instruction *I) {
     auto *PHI = cast<PHINode>(I);
     SmallVector<Value *, 4> NegatedIncomingValues(PHI->getNumOperands());
     for (auto I : zip(PHI->incoming_values(), NegatedIncomingValues)) {
+      // Don't negate indvars to avoid infinite loops.
+      if (DT.dominates(PHI, std::get<0>(I)))
```

Comment:
```suggestion
      if (DT.dominates(PHI->getParent(), std::get<0>(I)))
```
Shouldn't make a difference here, but I don't think using value dominance really makes sense in this context.